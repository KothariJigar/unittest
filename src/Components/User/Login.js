import React, { useState } from "react";
import { useNavigate } from "react-router-dom";

export default function Login() {
  const [name, setUser] = useState("");
  const [pass, setPass] = useState("");
  const [error, setError] = useState("");
  const navigate = useNavigate();

  const handleLogin = (e) => {
    e.preventDefault();
    const loginDetails = JSON.parse(localStorage.getItem("data"));
    console.log("7844", loginDetails);
    if (loginDetails?.name === name && loginDetails?.password === pass) {
      setError("login successfully");
      //navigate("/UserDetailsEditPage");
      navigate("/UserDetailsPage");
    }
  };

  return (
    <>
      <div>
        <h1>LoggedIn User </h1>
      </div>
      {error && <p>{error}</p>}
      <form>
        <label className="label">UserName</label>
        <input
          className="input"
          type="text"
          value={name}
          onChange={(e) => setUser(e.target.value)}
          data-testid="input-field"
        />
        <label className="label">password</label>
        <input
          className="input"
          type="password"
          value={pass}
          onChange={(e) => setPass(e.target.value)}
          data-testid="password-field"
        />
        <button className="btn" onClick={(e) => handleLogin(e)}>
          Login
        </button>
      </form>
    </>
  );
}
