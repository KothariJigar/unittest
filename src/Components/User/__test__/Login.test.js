import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import Login from "../Login";

beforeEach(() => {
  render(
    <BrowserRouter>
      <Login />
    </BrowserRouter>
  );
});

describe("Login component tests", () => {
  it("should render login page with two input fields and a button", () => {
    const username = screen.getByTestId("input-field");
    const password = screen.getByTestId("password-field");
    const loginButton = screen.getByRole("button");

    expect(username).toBeInTheDocument();
    expect(password).toBeInTheDocument();
    expect(loginButton).toBeInTheDocument();
  });

  it("should show username value on change", () => {
    const username = screen.getByTestId("input-field");
    fireEvent.change(username, { target: { value: "test" } });
    expect(username).toHaveValue("test");
  });

  it("should show password value on change", () => {
    const username = screen.getByTestId("password-field");
    fireEvent.change(username, { target: { value: "passwordtest" } });
    expect(username).toHaveValue("passwordtest");
  });

  it("should login", () => {
    const loginButton = screen.getByRole("button");
    fireEvent.click(loginButton);
  });

  it("should match snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
afterEach(cleanup);
