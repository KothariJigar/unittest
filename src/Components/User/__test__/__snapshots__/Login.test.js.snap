// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component tests should match snapshot 1`] = `
<div>
  <div>
    <h1>
      LoggedIn User 
    </h1>
  </div>
  <form>
    <label
      class="label"
    >
      UserName
    </label>
    <input
      class="input"
      data-testid="input-field"
      type="text"
      value=""
    />
    <label
      class="label"
    >
      password
    </label>
    <input
      class="input"
      data-testid="password-field"
      type="password"
      value=""
    />
    <button
      class="btn"
    >
      Login
    </button>
  </form>
</div>
`;
